net/samba4x/Makefile

```diff
@@ -0,0 +1,273 @@
+endif
+ifndef CONFIG_SAMBA4X_IPRINT
+  SAMBA4X_OPTIONS += --disable-iprint
+endif
+ifndef CONFIG_SAMBA4X_PTHREADPOOL
+  SAMBA4X_OPTIONS += --disable-pthreadpool
+endif
+ifndef CONFIG_SAMBA4X_PYTHON
+  SAMBA4X_OPTIONS += --disable-python
+endif
+ifndef CONFIG_SAMBA4X_RPATH
+  SAMBA4X_OPTIONS += --disable-rpath
+endif
+ifndef CONFIG_SAMBA4X_RPATH_INSTALL
+  SAMBA4X_OPTIONS += --disable-rpath-install
+endif
+ifndef CONFIG_SAMBA4X_RPATH_PRIVATE_INSTALL
+  SAMBA4X_OPTIONS += --disable-rpath-private-install
+endif
+ifndef CONFIG_SAMBA4X_ACL
+  SAMBA4X_OPTIONS += --without-acl-support
+endif
+ifndef CONFIG_SAMBA4X_ADS
+  SAMBA4X_OPTIONS += --without-ads
+endif
+ifndef CONFIG_SAMBA4X_AD_DC
+  SAMBA4X_OPTIONS += --without-ad-dc
+endif
+ifndef CONFIG_SAMBA4X_AUTOMOUNT
+  SAMBA4X_OPTIONS += --without-automount
+endif
+ifndef CONFIG_SAMBA4X_DMAPI
+  SAMBA4X_OPTIONS += --without-dmapi
+endif
+ifndef CONFIG_SAMBA4X_DNSUPDATE
+  SAMBA4X_OPTIONS += --without-dnsupdate
+endif
+ifndef CONFIG_SAMBA4X_FAM
+  SAMBA4X_OPTIONS += --without-fam
+endif
+ifndef CONFIG_SAMBA4X_GPGME
+  SAMBA4X_OPTIONS += --without-gpgme
+endif
+ifndef CONFIG_SAMBA4X_ICONV
+  SAMBA4X_OPTIONS += --without-iconv
+endif
+ifndef CONFIG_SAMBA4X_LDAP
+  SAMBA4X_OPTIONS += --without-ldap
+endif
+ifndef CONFIG_SAMBA4X_LIBARCHIVE
+  SAMBA4X_OPTIONS += --without-libarchive
+endif
+ifndef CONFIG_SAMBA4X_LTTNG
+  SAMBA4X_OPTIONS += --without-lttng
+endif
+ifndef CONFIG_SAMBA4X_PAM
+  SAMBA4X_OPTIONS += --without-pam
+endif
+ifndef CONFIG_SAMBA4X_QUOTAS
+  SAMBA4X_OPTIONS += --without-quotas
+endif
+ifndef CONFIG_SAMBA4X_REGEDIT
+  SAMBA4X_OPTIONS += --without-regedit
+endif
+ifndef CONFIG_SAMBA4X_RELRO
+  SAMBA4X_OPTIONS += --without-relro
+endif
+ifndef CONFIG_SAMBA4X_SYSTEMD
+  SAMBA4X_OPTIONS += --without-systemd
+endif
+ifndef CONFIG_SAMBA4X_UTMP
+  SAMBA4X_OPTIONS += --without-utmp
+endif
+ifndef CONFIG_SAMBA4X_WINBIND
+  SAMBA4X_OPTIONS += --without-winbind
+endif
+
+CONFIGURE_ARGS += \
+		--hostcc="$(shell which gcc)" \
+		--bundled-libraries="$(SAMBA_BUNDLED_LIBS)" \
+		--cross-compile \
+		--cross-answers="$(PKG_BUILD_DIR)/cache.txt" \
+		--accel-aes=none \
+		--enable-fhs \
+		--fatal-errors \
+		--pedantic \
+		--without-ntvfs-fileserver \
+		$(SAMBA4X_OPTIONS) \
+		--nopyc --nopyo
+
+define Build/Configure
+	$(CP) ./files/cache.txt $(PKG_BUILD_DIR)/cache.txt;
+	echo -e "\nChecking uname sysname type: \"$(VERSION_DIST)\" \
+	\nChecking uname release type: \"$(LINUX_VERSION)\" \
+	\nChecking uname machine type: \"$(ARCH)\" \
+	\nChecking uname version type: \"$(VERSION_DIST) Linux-$(LINUX_VERSION) $(shell date +%Y-%m-%d)\"\n" >> $(PKG_BUILD_DIR)/cache.txt;
+
+	$(call Build/Configure/Default,configure)
+endef
+
+TARGET_CFLAGS += \
+	-fdata-sections \
+	-ffunction-sections
+
+TARGET_LDFLAGS += \
+	-Wl,--gc-sections
+
+define Build/Compile
+	(cd $(PKG_BUILD_DIR); \
+		./buildtools/bin/waf build -vvv \
+		--targets=smbd/smbd,nmbd,smbpasswd,smbclient,nmblookup,net \
+		-j$(shell nproc))
```

Comment:
Wouldn't it be better to use `$(PKG_JOBS)` here?

Reply:
thanks, it will be fixed for next commit.

Reply:
Just noticed, do you get a smbclient binary? I had to change this to ```client/smbclient``` since that's the waf target name.